packages/vite/src/types/http-proxy.d.ts

```diff
@@ -212,6 +212,8 @@ export namespace HttpProxy {
     localAddress?: string | undefined
     /** Changes the origin of the host header to the target URL. */
     changeOrigin?: boolean | undefined
+    /** Rewrites the Origin header of a WebSocket request to match the the target */
+    rewriteWsOrigin?: boolean | undefined
```

Comment:
Should extensions of the package be added here?

https://github.com/vitejs/vite/blob/61357f67cdb8eca2c551150a1f0329e272f4da62/packages/vite/src/node/server/middlewares/proxy.ts#L13

which the above is also linked by the docs.

Reply:
Good point, should probably be a separate interface definition.

Reply:
https://github.com/vitejs/vite/pull/17563/commits/38d0d6ce94dda287e1861b8fe07ff2138141f425